src/SwedbankPay.Sdk.Infrastructure/CancelTransactionDto.cs

```diff
@@ -0,0 +1,47 @@
+﻿using SwedbankPay.Sdk.PaymentInstruments.Invoice;
+using SwedbankPay.Sdk.PaymentInstruments.Trustly;
+using SwedbankPay.Sdk.PaymentOrders;
+
+namespace SwedbankPay.Sdk
+{
+    internal class CancelTransactionDto
+    {
+        public CancelTransactionDto(ICancelTransaction transaction)
+        {
+            Description = transaction.Description;
+            PayeeReference = transaction.PayeeReference;
+            TransactionActivity = transaction.TransactionActivity.Value;
+        }
+
+        public CancelTransactionDto(PaymentInstruments.MobilePay.CancelTransaction transaction)
+        {
+            Description = transaction.Description;
+            PayeeReference = transaction.PayeeReference;
+        }
+
+        public CancelTransactionDto(TrustlyPaymentCancelTransaction transaction)
+        {
+            Description = transaction.Description;
+            PayeeReference = transaction.PayeeReference;
+        }
+
+        public CancelTransactionDto(PaymentInstruments.Vipps.CancelTransaction transaction)
+        {
+            Description = transaction.Description;
+            PayeeReference = transaction.PayeeReference;
+        }
+
+        public CancelTransactionDto(PaymentOrderCancelTransaction transaction)
+        {
+
+            Description = transaction.Description;
+            PayeeReference = transaction.PayeeReference;
+        }
```

Comment:
Sure, 4 out of 5 is identical. So it could be reduced to two constructors. How to reduce it to one I do not know unless you don't want `TransactionActivity` for card payments, or if every other payment instrument should now expose it and we hope the api let us send in extra properties.

Reply:
Perhaps card payments support `TransactionActivity` too? Do we know? Let's find out! :)